pyqtgraph/functions.py

```diff
@@ -288,16 +285,14 @@ def mkColor(*args):
         else:
             raise TypeError(err)
     elif len(args) == 3:
-        (r, g, b) = args
+        r, g, b = args
         a = 255
     elif len(args) == 4:
-        (r, g, b, a) = args
+        r, g, b, a = args
     else:
         raise TypeError(err)
-    
-    args = [r,g,b,a]
-    args = [0 if np.isnan(a) or np.isinf(a) else a for a in args]
-    args = list(map(int, args))
+
+    args = np.nan_to_num((r, g, b, a), nan=0, posinf=255, neginf=0).astype(int).tolist()
```

Comment:
so `args = [int(a) if math.isfinite(a) else 0 for a in args]` doesn't work due to a case with PySide2 5.12 having integer overflow when passing in one of the odd numpy values (like `np.inf` or something like that).  What _could_ work is `[int(a) if np.isfinite(a) else 0 for a in args]` but what I didn't like about that was I was using a numpy function that is meant to take in array-like fields, but on a scalar.  This PR aims to break that pattern throughout the entire library.

the `np.nan_to_num` seemed to be a good compromise.

Although since you brought this up,  I tested the unpacking, which is why I converted back to a list, but turns out I actually don't need to do that, that works on a numpy array too

```python
>>> def test(a, b, c, *args):
>>>     print(a, b, c, args)
>>> array = np.arange(10)
>>> test(*array)
0 1 2 (3, 4, 5, 6, 7, 8, 9)
```

This means this can be simplified a bit by ditching the `.tolist()`

Reply:
Some time ago, I found that unpacking an ndarray is slower than calling tolist() and then unpacking that. Probably ndarray unpacking is implemented as a separate function call per element.

Regarding np.isfinite() vs math.isfinite(), the difference is that np.isfinite() operates on floats so will first convert the input to a float. This matters for big integers. So:
```python
In [28]: type(2**64)
Out[28]: int

In [29]: math.isfinite(2**64)
Out[29]: True

In [30]: np.isfinite(2**64)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-30-ffa69f841cb0> in <module>
----> 1 np.isfinite(2**64)

TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

Reply:
Thanks for that explanation! That's really good to know.

Reply:
Okay, one more try:
``` python
args = [
    fn.clip_scalar( int(a), 0, 255) 
    if math.isfinite(a) else 0 
    for a in args]
```
Makes it clear that we are doing input conditioning for mkColor.
And I don't think that it does a siginificantly larger number of checks and calculations than the other approaches.

Reply:
I think this would fail the CI. There is a test that tests for failure with the value 1e129. It ends up entering this branch:
```python
        elif np.issubdtype(type(args[0]), np.floating):
            r = g = b = int(args[0] * 255)
            a = 255
```
So, r, g, b are now big Python ints which would get clipped to [0, 255] and get accepted instead.
By the way, it's this transformation to big Python int that's causing all this trouble.

Reply:
I'm going to merge this PR as is, as it works w/o generating errors on any of the supported combinations; of course, if someone identifies a faster / more robust method to unpack safely, a PR would be appreciated 👍🏻